### PR TITLE
Unwrap ChainerX connected array from `Variable`

### LINKED ISSUE #6279: Get chainerx.ndarray with nodes from Variable
Support optimized backproppable calculation by unwrapping chainerx variables and directly manipulate arrays.

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #6279 
Introduces `Variable.chx_view()` which returns a view (`chx.ndarray`) of the original array.

### CODE REVIEW

File: chainer/variable.py [hunk 1]
```diff
@@ -867,6 +867,24 @@ def array(self, d):
             self._data[0] = d
             self._has_chainerx_array = False
 
+    def chx_view(self):
```
Comment (hvy) [inline chainer/variable.py]: Could you clarify why it's returning a view?

Comment (niboshi) [reply chainer/variable.py]: If it is not, `var.chx_view().require_grad()` would affect the original array, making `var` in an invalid state.

Comment (hvy) [reply chainer/variable.py]: Thanks for the explanation.

Comment (hvy) [inline chainer/variable.py]: How about not abbreviating chainerx -> chx and name it `chainerx_view`? This is more consistent with other APIs.

Comment (niboshi) [reply chainer/variable.py]: Changed to `chainex_array` after discussion. We may consider changing `chainerx` to `chx` throughout the Chainer code afterwards (e.g. `to_chainerx` -> `to_chx`).

**Comment by hvy** [comment]
jenkins, test this please.

File: tests/chainer_tests/test_variable.py [hunk 2]
```diff
@@ -2968,4 +2968,84 @@ def test_backward_cpu_lazy_grad_sum(self):
...
+        else:
+            y_arr.backward()
+
+        # x.grad is set
+        assert x.grad is not None
+        chainerx.testing.assert_array_equal_ex(
+            chainerx.array([2, 4], np.float32), x.grad)
+
+
+class TestVariableChainerxArrayView(unittest.TestCase):
```
Comment (hvy) [inline tests/chainer_tests/test_variable.py]: We need the ChainerX attribute `@attr.chainerx` here.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
CI failures seem unrelated. LGTM!

**Pull request merged**